checker/checker_asset.go

```diff
@@ -27,34 +26,18 @@ import (
 )
 
 type assetChecker struct {
-	assetPath string
-	cfgYML    string
+	assetPath       string
+	cfgYML          string
+	checkerType     okgo.CheckerType
+	checkerPriority okgo.CheckerPriority
```

Comment:
Rather than requiring these to be initialized on construction externally, I think it makes more sense to have these as cached values populated by the functions once.

Since we care about both value and error, I think simplest is to store functions as "once" variants, where calling methods sets the variable value to the "once" variant if it's `nil` and then just invokes it:

```suggestion
	// typeFnOnce runs the logic for computing the type exactly once (subsequent calls return the cached value).
	// initTypeFnOnce ensures that this function is initialized only once.
	initTypeFnOnce sync.Once
	typeFnOnce     func() (okgo.CheckerType, error)

	// priorityFnOnce runs the logic for computing the priority exactly once (subsequent calls return the cached value).
	// initPriorityFnOnce ensures that this function is initialized only once.
	initPriorityFnOnce sync.Once
	priorityFnOnce     func() (okgo.CheckerPriority, error)
```